Services/Concurrency/RateLimiting.cs

```diff
@@ -58,6 +60,50 @@ public class RateLimiting : IRateLimiting
             log.Info("Rate limiting started. This message should appear only once in the logs.");
         }
 
+        public Models.Simulation.SimulationRateLimits GetRateLimits()
+        {
+            return new Models.Simulation.SimulationRateLimits
+            {
+                ConnectionsPerSecond = this.connections.EventsPerTimeUnit,
+                RegistryOperationsPerMinute = this.registryOperations.EventsPerTimeUnit,
+                TwinWritesPerSecond = this.twinReads.EventsPerTimeUnit,
+                TwinReadsPerSecond = this.twinWrites.EventsPerTimeUnit,
+                DeviceMessagesPerSecond = this.messaging.EventsPerTimeUnit
+            };
+        }
+
+        public void SetCounters(
+            Models.Simulation.SimulationRateLimits simulationRateLimits, 
+            ILogger log)
+        {
+            if (simulationRateLimits.ConnectionsPerSecond > 0)
+            {
+                this.connections = new PerSecondCounter(simulationRateLimits.ConnectionsPerSecond, "Device connections", log);
```

Comment:
How should we handle cases where if condition is false ? Any default values for counters ?